gtfs/spec/en/reference.md

```diff
@@ -32,6 +33,22 @@ This section defines terms that are used throughout this document.
 * **Field optional** - The field column may be omitted from your feed. If you choose to include an optional column, each record in your feed must have a value for that column. You may include an empty string as a value for records that do not have values for the column. Some optional fields permit an empty string as a value. To enter an empty string, just omit any text between the commas for that field. Note that 0 is interpreted as "a string of value 0", and is not an empty string.
 * **Dataset unique** - The field contains a value that maps to a single distinct entity within the column. For example, if a route is assigned the ID **1A**, then no other route may use that route ID. However, you may assign the ID **1A** to a location because locations are a different type of entity than routes.
 
+## Field Types
+
+- **Ad-hoc** - Please see the detail column of the field for the definition of the field type.
```

Comment:
Which fields would this describe?  Without knowing that, I think `Misc` is probably a better value.

Reply:
I've mainly done it for `currency_type`(ISO 4217) and `agency_phone` (which should be E123 but currently isn't specified as such).

But to answer your question I've checked every field and it could also be used for:
- agency_email, feed_contact_email (but I can add a type for that now that I see there are two of them)
- shape_dist_traveled (positive float)
- price (positive float)
- feed_version (not an ID, not really a Text string)

Reply:
I must confess I don't like "Misc", it sounds as "what we haven't sorted". Ad-hoc is distinct, it is saying that the format is distinct.

But if another anglo endorse the "Misc", I'll go for it. @antrim?

Reply:
>I must confess I don't like "Misc", it sounds as "what we haven't sorted".

Isn't that what it is :)?

I'm not married to "Misc", but "Ad-Hoc" definitely doesn't sound right.  To me in English "Ad-Hoc" implies something that's a "one-off" or something done once for a particular purpose without organization.  To me it almost implies that you could put whatever you want in that field.

Thinking more, I'd actually vote to remove this type entirely and just list all types here, even if they only occur once in the spec.

Reply:
I'm adding the Non-negative float type (roll eyes) and the email type.

Reply:
Regarding ad-hoc, I'm just gonna remove it.